### ISSUE TITLE
Registration of provisioned Host is never retried

### ISSUE THREAD

**Issue by zaneb** [metal3-io/baremetal-operator#698]
When calling `Adopt()` (in the provisioned or externally provisioned states), we assume that either the Node does not exist in Ironic (in which case we attempt to register it), or that it is in the manageable state. If the Node exists in the enrol state due to Ironic having failed to move it to manageable (e.g. because the BMC is temporarily unreachable), then we go into an infinite reconcile loop where we just complain about the wrong state and never attempt to try again:

```
2020-10-21T18:13:59.447 Reconciling BareMetalHost {Request.Namespace: 'openshift-machine-api', Request.Name: 'openshift-master-0-0'}
2020-10-21T18:13:59.447 verifying ironic provisioner dependencies {host: 'openshift-master-0-0'}
2020-10-21T18:13:59.530 found existing node by ID {host: 'openshift-master-0-0'}
2020-10-21T18:13:59.531 Reconciler error {controller: 'metal3-baremetalhost-controller', request: 'openshift-machine-api/openshift-master-0-0', error: 'action "externally provisioned" failed: Invalid state for adopt: enroll'}
github.com/go-logr/zapr.(*zapLogger).Error
        /go/src/github.com/metal3-io/baremetal-operator/vendor/github.com/go-logr/zapr/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:258
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
        /go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
        /go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
        /go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
        /go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
```

Original bug report: https://bugzilla.redhat.com/show_bug.cgi?id=1890256

**Comment by zaneb** [comment]
I have a patch that I think may help with this, but I'm waiting for #388 to merge

@furkatgofurov7 this may be related to some of the issues you're seeing. #697 is a similar example of not handling all the possible states in `Adopt()`.

**Comment by furkatgofurov7** [comment]
@zaneb agree. Will the patch that you mentioned above, cover the case of #697 as well?

**Comment by zaneb** [comment]
No, we'll need a separate fix for #697.
